### PR TITLE
#162171012 Setup unit testing

### PR DESCRIPTION
**Pull request by henperi**
What does this PR do?
- It sets up and integrates unit testing functionality to the project.

Description of Task to be completed?
- Install chai
- Install chai-http
- Install mocha
- Install mocha-lcov-reporter
- Create test/index.js as the test entry point
- Create test/users.js, test/likes.js, test/comments.js, test/articles.js
- Write a test script to run the unit tests in test folder through the test/index.js

How should this be manually tested?
- Clone this project
- ```npm install```
- ```npm test```

Any background context you want to provide?
Nil

What are the relevant pivotal tracker stories?
[Finishes #162171012]

Screenshots (if appropriate)
Nil

Questions:
Nil

### CODE REVIEW

File: .nyc_output/5c49856d-857e-44d9-b782-215413b42cb7.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Chidi3b3r3) [inline .nyc_output/5c49856d-857e-44d9-b782-215413b42cb7.json]: Please add .nyc_output to our .gitignore file

File: package.json [hunk 2]
```diff
@@ -4,12 +4,15 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "nyc --reporter=lcov mocha ./test/index.js --require babel-register --exit"
```
Comment (Chidi3b3r3) [inline package.json]: Have a separate script for generating coverage report and another script for running your tests.
Something along "test:coverage": "...." and "test": "..."

Comment (henperi) [reply package.json]: Thanks for your feedback. I've removed the coverage line from the test script.

File: .nyc_output/5c49856d-857e-44d9-b782-215413b42cb7.json (hunk 1 above)
Comment (henperi) [reply .nyc_output/5c49856d-857e-44d9-b782-215413b42cb7.json]: Thanks for your feedback. I just added .nyc_output to gitignore file

**Pull request merged**